sql/hive-thriftserver/pom.xml

```diff
@@ -143,6 +143,10 @@
       <groupId>commons-cli</groupId>
       <artifactId>commons-cli</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.apache.httpcomponents</groupId>
+      <artifactId>httpcore</artifactId>
+    </dependency>
```

Comment:
Explicit declaration dependencies(org.apache.httpcomponents:httpcore) in the pom file of sub-module(spark-hive-thriftserver)

Reply:
If we do not explicit import the dependency, `Scala 2.13 build with SBT` will fail to compile
<img width="1360" alt="image" src="https://user-images.githubusercontent.com/15246973/193257971-ccb75362-8344-4b16-9c00-cec6957a9999.png">

Reply:
<img width="815" alt="image" src="https://user-images.githubusercontent.com/15246973/193258253-11c2a7e9-b5b4-4c48-a8bc-958720c380ae.png">

Reply:
Which plugin upgrade caused this problem?

Reply:
maven-dependency-plugin

Reply:
But I found no difference in the results of `dependency:tree`

**311**
```
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ spark-hive-thriftserver_2.13 ---
[INFO] org.apache.spark:spark-hive-thriftserver_2.13:jar:3.4.0-SNAPSHOT
[INFO] +- org.apache.spark:spark-hive_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-core_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-sql_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.hive:hive-common:jar:2.3.9:compile
[INFO] |  |  +- org.apache.commons:commons-compress:jar:1.21:compile
[INFO] |  |  +- com.tdunning:json:jar:1.8:compile
[INFO] |  |  \- com.github.joshelser:dropwizard-metrics-hadoop-metrics2-reporter:jar:0.1.2:compile
[INFO] |  +- org.apache.hive:hive-exec:jar:core:2.3.9:compile
[INFO] |  |  +- org.apache.hive:hive-vector-code-gen:jar:2.3.9:compile
[INFO] |  |  |  \- org.apache.velocity:velocity:jar:1.5:compile
[INFO] |  |  +- org.antlr:antlr-runtime:jar:3.5.2:compile
[INFO] |  |  +- org.antlr:ST4:jar:4.0.4:compile
[INFO] |  |  +- com.google.code.gson:gson:jar:2.2.4:compile
[INFO] |  |  \- stax:stax-api:jar:1.0.1:compile
[INFO] |  +- org.apache.hive:hive-metastore:jar:2.3.9:compile
[INFO] |  |  +- javolution:javolution:jar:5.5.1:compile
[INFO] |  |  +- com.google.protobuf:protobuf-java:jar:2.5.0:compile
[INFO] |  |  +- com.jolbox:bonecp:jar:0.8.0.RELEASE:compile
[INFO] |  |  +- com.zaxxer:HikariCP:jar:2.5.1:compile
[INFO] |  |  +- org.datanucleus:datanucleus-api-jdo:jar:4.2.4:compile
[INFO] |  |  +- org.datanucleus:datanucleus-rdbms:jar:4.1.19:compile
[INFO] |  |  +- commons-pool:commons-pool:jar:1.5.4:compile
[INFO] |  |  +- commons-dbcp:commons-dbcp:jar:1.4:compile
[INFO] |  |  +- javax.jdo:jdo-api:jar:3.0.1:compile
[INFO] |  |  |  \- javax.transaction:jta:jar:1.1:compile
[INFO] |  |  \- org.datanucleus:javax.jdo:jar:3.2.0-m3:compile
[INFO] |  |     \- javax.transaction:transaction-api:jar:1.1:compile
[INFO] |  +- org.apache.hive:hive-serde:jar:2.3.9:compile
[INFO] |  |  \- net.sf.opencsv:opencsv:jar:2.3:compile
[INFO] |  +- org.apache.hive:hive-shims:jar:2.3.9:compile
[INFO] |  |  +- org.apache.hive.shims:hive-shims-common:jar:2.3.9:compile
[INFO] |  |  +- org.apache.hive.shims:hive-shims-0.23:jar:2.3.9:runtime
[INFO] |  |  \- org.apache.hive.shims:hive-shims-scheduler:jar:2.3.9:runtime
[INFO] |  +- org.apache.hive:hive-llap-common:jar:2.3.9:compile
[INFO] |  +- org.apache.hive:hive-llap-client:jar:2.3.9:test
[INFO] |  +- org.apache.avro:avro:jar:1.11.1:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.13.4:compile
[INFO] |  +- org.apache.avro:avro-mapred:jar:1.11.1:compile
[INFO] |  |  \- org.apache.avro:avro-ipc:jar:1.11.1:compile
[INFO] |  |     \- org.tukaani:xz:jar:1.8:compile
[INFO] |  +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO] |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.13:compile
[INFO] |  |  \- org.codehaus.jackson:jackson-core-asl:jar:1.9.13:compile
[INFO] |  +- commons-codec:commons-codec:jar:1.15:compile
[INFO] |  +- joda-time:joda-time:jar:2.10.14:compile
[INFO] |  +- org.jodd:jodd-core:jar:3.5.2:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.0:compile
[INFO] |  +- org.datanucleus:datanucleus-core:jar:4.1.17:compile
[INFO] |  +- org.apache.hadoop:hadoop-client-runtime:jar:3.3.4:compile
[INFO] |  +- org.apache.thrift:libthrift:jar:0.12.0:compile
[INFO] |  +- org.apache.thrift:libfb303:jar:0.9.3:compile
[INFO] |  \- org.apache.derby:derby:jar:10.14.2.0:compile
[INFO] +- org.apache.spark:spark-core_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] |  +- com.twitter:chill_2.13:jar:0.10.0:compile
[INFO] |  |  \- com.esotericsoftware:kryo-shaded:jar:4.0.2:compile
[INFO] |  |     \- com.esotericsoftware:minlog:jar:1.3.0:compile
[INFO] |  +- com.twitter:chill-java:jar:0.10.0:compile
[INFO] |  +- org.apache.xbean:xbean-asm9-shaded:jar:4.21:compile
[INFO] |  +- org.apache.hadoop:hadoop-client-api:jar:3.3.4:compile
[INFO] |  +- org.apache.spark:spark-launcher_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-kvstore_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  |  +- org.fusesource.leveldbjni:leveldbjni-all:jar:1.8:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.13.4:compile
[INFO] |  +- org.apache.spark:spark-network-common_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  |  \- com.google.crypto.tink:tink:jar:1.7.0:compile
[INFO] |  +- org.apache.spark:spark-network-shuffle_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-unsafe_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- javax.activation:activation:jar:1.1.1:compile
[INFO] |  +- org.apache.curator:curator-recipes:jar:2.13.0:compile
[INFO] |  |  \- org.apache.curator:curator-framework:jar:2.13.0:compile
[INFO] |  |     \- org.apache.curator:curator-client:jar:2.13.0:compile
[INFO] |  +- org.apache.zookeeper:zookeeper:jar:3.6.2:compile
[INFO] |  |  +- org.apache.zookeeper:zookeeper-jute:jar:3.6.2:compile
[INFO] |  |  \- org.apache.yetus:audience-annotations:jar:0.5.0:compile
[INFO] |  +- jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.12.0:compile
[INFO] |  +- org.apache.commons:commons-math3:jar:3.6.1:compile
[INFO] |  +- org.apache.commons:commons-text:jar:1.9:compile
[INFO] |  +- commons-io:commons-io:jar:2.11.0:compile
[INFO] |  +- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] |  +- org.apache.commons:commons-collections4:jar:4.4:compile
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.36:compile
[INFO] |  +- org.slf4j:jul-to-slf4j:jar:1.7.36:compile
[INFO] |  +- org.slf4j:jcl-over-slf4j:jar:1.7.36:compile
[INFO] |  +- org.apache.logging.log4j:log4j-slf4j-impl:jar:2.19.0:compile
[INFO] |  +- org.apache.logging.log4j:log4j-api:jar:2.19.0:compile
[INFO] |  +- org.apache.logging.log4j:log4j-core:jar:2.19.0:compile
[INFO] |  +- org.apache.logging.log4j:log4j-1.2-api:jar:2.19.0:compile
[INFO] |  +- com.ning:compress-lzf:jar:1.1:compile
[INFO] |  +- org.xerial.snappy:snappy-java:jar:1.1.8.4:compile
[INFO] |  +- org.lz4:lz4-java:jar:1.8.0:compile
[INFO] |  +- com.github.luben:zstd-jni:jar:1.5.2-3:compile
[INFO] |  +- org.roaringbitmap:RoaringBitmap:jar:0.9.32:compile
[INFO] |  |  \- org.roaringbitmap:shims:jar:0.9.32:runtime
[INFO] |  +- org.scala-lang.modules:scala-xml_2.13:jar:2.1.0:compile
[INFO] |  +- org.scala-lang:scala-library:jar:2.13.8:compile
[INFO] |  +- org.scala-lang:scala-reflect:jar:2.13.8:compile
[INFO] |  +- org.json4s:json4s-jackson_2.13:jar:3.7.0-M11:compile
[INFO] |  |  \- org.json4s:json4s-core_2.13:jar:3.7.0-M11:compile
[INFO] |  |     +- org.json4s:json4s-ast_2.13:jar:3.7.0-M11:compile
[INFO] |  |     \- org.json4s:json4s-scalap_2.13:jar:3.7.0-M11:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-client:jar:2.36:compile
[INFO] |  |  +- jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile
[INFO] |  |  \- org.glassfish.hk2.external:jakarta.inject:jar:2.6.1:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-common:jar:2.36:compile
[INFO] |  |  +- jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile
[INFO] |  |  \- org.glassfish.hk2:osgi-resource-locator:jar:1.0.3:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-server:jar:2.36:compile
[INFO] |  |  \- jakarta.validation:jakarta.validation-api:jar:2.0.2:compile
[INFO] |  +- org.glassfish.jersey.containers:jersey-container-servlet:jar:2.36:compile
[INFO] |  +- org.glassfish.jersey.containers:jersey-container-servlet-core:jar:2.36:compile
[INFO] |  +- org.glassfish.jersey.inject:jersey-hk2:jar:2.36:compile
[INFO] |  |  +- org.glassfish.hk2:hk2-locator:jar:2.6.1:compile
[INFO] |  |  |  +- org.glassfish.hk2.external:aopalliance-repackaged:jar:2.6.1:compile
[INFO] |  |  |  +- org.glassfish.hk2:hk2-api:jar:2.6.1:compile
[INFO] |  |  |  \- org.glassfish.hk2:hk2-utils:jar:2.6.1:compile
[INFO] |  |  \- org.javassist:javassist:jar:3.25.0-GA:compile
[INFO] |  +- io.netty:netty-all:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-buffer:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-codec:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-common:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-handler:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-transport-native-unix-common:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-resolver:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-transport:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-transport-classes-epoll:jar:4.1.80.Final:compile
[INFO] |  |  \- io.netty:netty-transport-classes-kqueue:jar:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-transport-native-epoll:jar:linux-aarch_64:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-transport-native-kqueue:jar:osx-aarch_64:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-tcnative-classes:jar:2.0.54.Final:compile
[INFO] |  +- com.clearspring.analytics:stream:jar:2.9.6:compile
[INFO] |  +- io.dropwizard.metrics:metrics-core:jar:4.2.10:compile
[INFO] |  +- io.dropwizard.metrics:metrics-jvm:jar:4.2.10:compile
[INFO] |  +- io.dropwizard.metrics:metrics-json:jar:4.2.10:compile
[INFO] |  +- io.dropwizard.metrics:metrics-graphite:jar:4.2.10:compile
[INFO] |  +- io.dropwizard.metrics:metrics-jmx:jar:4.2.10:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.4:compile
[INFO] |  +- com.fasterxml.jackson.module:jackson-module-scala_2.13:jar:2.13.4:compile
[INFO] |  |  \- com.thoughtworks.paranamer:paranamer:jar:2.8:compile
[INFO] |  +- org.apache.ivy:ivy:jar:2.5.0:compile
[INFO] |  +- oro:oro:jar:2.0.8:compile
[INFO] |  +- net.razorvine:pickle:jar:1.2:compile
[INFO] |  +- net.sf.py4j:py4j:jar:0.10.9.7:compile
[INFO] |  \- org.apache.commons:commons-crypto:jar:1.1.0:compile
[INFO] +- org.apache.spark:spark-catalyst_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] |  +- org.scala-lang.modules:scala-parser-combinators_2.13:jar:1.1.2:compile
[INFO] |  +- org.apache.spark:spark-sketch_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.codehaus.janino:janino:jar:3.1.7:compile
[INFO] |  +- org.codehaus.janino:commons-compiler:jar:3.1.7:compile
[INFO] |  +- org.antlr:antlr4-runtime:jar:4.9.3:compile
[INFO] |  +- com.univocity:univocity-parsers:jar:2.9.1:compile
[INFO] |  +- org.apache.arrow:arrow-vector:jar:9.0.0:compile
[INFO] |  |  +- org.apache.arrow:arrow-format:jar:9.0.0:compile
[INFO] |  |  +- org.apache.arrow:arrow-memory-core:jar:9.0.0:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.13.4:compile
[INFO] |  |  \- com.google.flatbuffers:flatbuffers-java:jar:1.12.0:compile
[INFO] |  \- org.apache.arrow:arrow-memory-netty:jar:9.0.0:compile
[INFO] +- org.apache.spark:spark-hive_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] +- org.scala-lang.modules:scala-parallel-collections_2.13:jar:1.0.3:compile
[INFO] +- com.google.guava:guava:jar:14.0.1:provided
[INFO] +- org.apache.hive:hive-cli:jar:2.3.9:compile
[INFO] |  +- commons-lang:commons-lang:jar:2.6:compile
[INFO] |  \- jline:jline:jar:2.14.6:compile
[INFO] +- org.apache.hive:hive-jdbc:jar:2.3.9:compile
[INFO] |  \- org.apache.httpcomponents:httpcore:jar:4.4.14:compile
[INFO] +- org.apache.hive:hive-beeline:jar:2.3.9:compile
[INFO] |  \- net.sf.supercsv:super-csv:jar:2.2.0:compile
[INFO] +- org.apache.hive:hive-service-rpc:jar:3.1.3:compile
[INFO] +- org.eclipse.jetty:jetty-server:jar:9.4.48.v20220622:provided
[INFO] |  +- javax.servlet:javax.servlet-api:jar:3.1.0:provided
[INFO] |  +- org.eclipse.jetty:jetty-http:jar:9.4.48.v20220622:provided
[INFO] |  |  \- org.eclipse.jetty:jetty-util:jar:9.4.48.v20220622:provided
[INFO] |  \- org.eclipse.jetty:jetty-io:jar:9.4.48.v20220622:provided
[INFO] +- org.eclipse.jetty:jetty-servlet:jar:9.4.48.v20220622:provided
[INFO] |  +- org.eclipse.jetty:jetty-security:jar:9.4.48.v20220622:provided
[INFO] |  \- org.eclipse.jetty:jetty-util-ajax:jar:9.4.48.v20220622:provided
[INFO] +- org.seleniumhq.selenium:selenium-java:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-api:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-chrome-driver:jar:4.2.2:test
[INFO] |  |  +- org.seleniumhq.selenium:selenium-chromium-driver:jar:4.2.2:test
[INFO] |  |  \- org.seleniumhq.selenium:selenium-json:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-devtools-v100:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-devtools-v101:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-devtools-v102:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-devtools-v85:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-edge-driver:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-firefox-driver:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-ie-driver:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-opera-driver:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-remote-driver:jar:4.2.2:test
[INFO] |  |  +- com.beust:jcommander:jar:1.82:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-api:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-context:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-exporter-logging:jar:1.14.0:test
[INFO] |  |  |  +- io.opentelemetry:opentelemetry-sdk-metrics:jar:1.14.0:test
[INFO] |  |  |  \- io.opentelemetry:opentelemetry-sdk-logs:jar:1.14.0-alpha:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk-common:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk-extension-autoconfigure:jar:1.14.0-alpha:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk-trace:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-semconv:jar:1.14.0-alpha:test
[INFO] |  |  +- io.ous:jtoml:jar:2.0.0:test
[INFO] |  |  +- org.apache.commons:commons-exec:jar:1.3:test
[INFO] |  |  +- org.asynchttpclient:async-http-client:jar:2.12.3:test
[INFO] |  |  |  +- org.asynchttpclient:async-http-client-netty-utils:jar:2.12.3:test
[INFO] |  |  |  +- org.reactivestreams:reactive-streams:jar:1.0.3:test
[INFO] |  |  |  +- com.typesafe.netty:netty-reactive-streams:jar:2.0.4:test
[INFO] |  |  |  \- com.sun.activation:jakarta.activation:jar:1.2.2:test
[INFO] |  |  \- org.seleniumhq.selenium:selenium-http:jar:4.2.2:test
[INFO] |  |     \- dev.failsafe:failsafe:jar:3.2.4:test
[INFO] |  +- org.seleniumhq.selenium:selenium-safari-driver:jar:4.2.2:test
[INFO] |  \- org.seleniumhq.selenium:selenium-support:jar:4.2.2:test
[INFO] +- org.seleniumhq.selenium:htmlunit-driver:jar:3.62.0:test
[INFO] |  \- net.sourceforge.htmlunit:htmlunit:jar:2.62.0:test
[INFO] |     +- xalan:xalan:jar:2.7.2:test
[INFO] |     |  \- xalan:serializer:jar:2.7.2:test
[INFO] |     +- org.apache.httpcomponents:httpmime:jar:4.5.13:test
[INFO] |     +- net.sourceforge.htmlunit:htmlunit-core-js:jar:2.62.0:test
[INFO] |     +- net.sourceforge.htmlunit:neko-htmlunit:jar:2.62.0:test
[INFO] |     |  \- xerces:xercesImpl:jar:2.12.2:test
[INFO] |     |     \- xml-apis:xml-apis:jar:1.4.01:test
[INFO] |     +- net.sourceforge.htmlunit:htmlunit-cssparser:jar:1.12.0:test
[INFO] |     +- commons-net:commons-net:jar:3.8.0:test
[INFO] |     +- org.brotli:dec:jar:0.1.2:test
[INFO] |     +- com.shapesecurity:salvation2:jar:3.0.0:test
[INFO] |     \- org.eclipse.jetty.websocket:websocket-client:jar:9.4.46.v20220331:test
[INFO] |        +- org.eclipse.jetty:jetty-client:jar:9.4.48.v20220622:provided
[INFO] |        \- org.eclipse.jetty.websocket:websocket-common:jar:9.4.46.v20220331:test
[INFO] |           \- org.eclipse.jetty.websocket:websocket-api:jar:9.4.46.v20220331:test
[INFO] +- org.apache.spark:spark-sql_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] |  +- org.rocksdb:rocksdbjni:jar:7.5.3:compile
[INFO] |  +- org.apache.spark:spark-catalyst_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.orc:orc-core:jar:shaded-protobuf:1.8.0:compile
[INFO] |  |  +- org.apache.orc:orc-shims:jar:1.8.0:compile
[INFO] |  |  +- io.airlift:aircompressor:jar:0.21:compile
[INFO] |  |  +- org.jetbrains:annotations:jar:17.0.0:compile
[INFO] |  |  \- org.threeten:threeten-extra:jar:1.7.1:compile
[INFO] |  +- org.apache.orc:orc-mapreduce:jar:shaded-protobuf:1.8.0:compile
[INFO] |  +- org.apache.hive:hive-storage-api:jar:2.7.3:compile
[INFO] |  +- org.apache.parquet:parquet-column:jar:1.12.3:compile
[INFO] |  |  +- org.apache.parquet:parquet-common:jar:1.12.3:compile
[INFO] |  |  \- org.apache.parquet:parquet-encoding:jar:1.12.3:compile
[INFO] |  \- org.apache.parquet:parquet-hadoop:jar:1.12.3:compile
[INFO] |     +- org.apache.parquet:parquet-format-structures:jar:1.12.3:compile
[INFO] |     \- org.apache.parquet:parquet-jackson:jar:1.12.3:runtime
[INFO] +- org.apache.spark:spark-tags_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] +- org.apache.spark:spark-tags_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] +- org.mockito:mockito-core:jar:4.6.1:test
[INFO] |  +- net.bytebuddy:byte-buddy:jar:1.12.10:test
[INFO] |  +- net.bytebuddy:byte-buddy-agent:jar:1.12.10:test
[INFO] |  \- org.objenesis:objenesis:jar:3.2:compile
[INFO] +- net.sf.jpam:jpam:jar:1.1:compile
[INFO] |  \- commons-logging:commons-logging:jar:1.1.3:compile
[INFO] +- commons-cli:commons-cli:jar:1.5.0:compile
[INFO] +- org.spark-project.spark:unused:jar:1.0.0:compile
[INFO] +- org.scalatest:scalatest_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-core_2.13:jar:3.2.13:test
[INFO] |  |  +- org.scalatest:scalatest-compatible:jar:3.2.13:test
[INFO] |  |  \- org.scalactic:scalactic_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-featurespec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-flatspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-freespec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-funsuite_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-funspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-propspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-refspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-wordspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-diagrams_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-matchers-core_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-shouldmatchers_2.13:jar:3.2.13:test
[INFO] |  \- org.scalatest:scalatest-mustmatchers_2.13:jar:3.2.13:test
[INFO] +- org.scalatestplus:scalacheck-1-16_2.13:jar:3.2.13.0:test
[INFO] |  \- org.scalacheck:scalacheck_2.13:jar:1.16.0:test
[INFO] +- org.scalatestplus:mockito-4-6_2.13:jar:3.2.13.0:test
[INFO] +- org.scalatestplus:selenium-4-2_2.13:jar:3.2.13.0:test
[INFO] +- junit:junit:jar:4.13.2:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] \- com.github.sbt:junit-interface:jar:0.13.3:test
[INFO]    \- org.scala-sbt:test-interface:jar:1.0:test
```

**330**
```
[INFO] --- maven-dependency-plugin:3.3.0:tree (default-cli) @ spark-hive-thriftserver_2.13 ---
[INFO] org.apache.spark:spark-hive-thriftserver_2.13:jar:3.4.0-SNAPSHOT
[INFO] +- org.apache.spark:spark-hive_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-core_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-sql_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.hive:hive-common:jar:2.3.9:compile
[INFO] |  |  +- org.apache.commons:commons-compress:jar:1.21:compile
[INFO] |  |  +- com.tdunning:json:jar:1.8:compile
[INFO] |  |  \- com.github.joshelser:dropwizard-metrics-hadoop-metrics2-reporter:jar:0.1.2:compile
[INFO] |  +- org.apache.hive:hive-exec:jar:core:2.3.9:compile
[INFO] |  |  +- org.apache.hive:hive-vector-code-gen:jar:2.3.9:compile
[INFO] |  |  |  \- org.apache.velocity:velocity:jar:1.5:compile
[INFO] |  |  +- org.antlr:antlr-runtime:jar:3.5.2:compile
[INFO] |  |  +- org.antlr:ST4:jar:4.0.4:compile
[INFO] |  |  +- com.google.code.gson:gson:jar:2.2.4:compile
[INFO] |  |  \- stax:stax-api:jar:1.0.1:compile
[INFO] |  +- org.apache.hive:hive-metastore:jar:2.3.9:compile
[INFO] |  |  +- javolution:javolution:jar:5.5.1:compile
[INFO] |  |  +- com.google.protobuf:protobuf-java:jar:2.5.0:compile
[INFO] |  |  +- com.jolbox:bonecp:jar:0.8.0.RELEASE:compile
[INFO] |  |  +- com.zaxxer:HikariCP:jar:2.5.1:compile
[INFO] |  |  +- org.datanucleus:datanucleus-api-jdo:jar:4.2.4:compile
[INFO] |  |  +- org.datanucleus:datanucleus-rdbms:jar:4.1.19:compile
[INFO] |  |  +- commons-pool:commons-pool:jar:1.5.4:compile
[INFO] |  |  +- commons-dbcp:commons-dbcp:jar:1.4:compile
[INFO] |  |  +- javax.jdo:jdo-api:jar:3.0.1:compile
[INFO] |  |  |  \- javax.transaction:jta:jar:1.1:compile
[INFO] |  |  \- org.datanucleus:javax.jdo:jar:3.2.0-m3:compile
[INFO] |  |     \- javax.transaction:transaction-api:jar:1.1:compile
[INFO] |  +- org.apache.hive:hive-serde:jar:2.3.9:compile
[INFO] |  |  \- net.sf.opencsv:opencsv:jar:2.3:compile
[INFO] |  +- org.apache.hive:hive-shims:jar:2.3.9:compile
[INFO] |  |  +- org.apache.hive.shims:hive-shims-common:jar:2.3.9:compile
[INFO] |  |  +- org.apache.hive.shims:hive-shims-0.23:jar:2.3.9:runtime
[INFO] |  |  \- org.apache.hive.shims:hive-shims-scheduler:jar:2.3.9:runtime
[INFO] |  +- org.apache.hive:hive-llap-common:jar:2.3.9:compile
[INFO] |  +- org.apache.hive:hive-llap-client:jar:2.3.9:test
[INFO] |  +- org.apache.avro:avro:jar:1.11.1:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.13.4:compile
[INFO] |  +- org.apache.avro:avro-mapred:jar:1.11.1:compile
[INFO] |  |  \- org.apache.avro:avro-ipc:jar:1.11.1:compile
[INFO] |  |     \- org.tukaani:xz:jar:1.8:compile
[INFO] |  +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO] |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.13:compile
[INFO] |  |  \- org.codehaus.jackson:jackson-core-asl:jar:1.9.13:compile
[INFO] |  +- commons-codec:commons-codec:jar:1.15:compile
[INFO] |  +- joda-time:joda-time:jar:2.10.14:compile
[INFO] |  +- org.jodd:jodd-core:jar:3.5.2:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.0:compile
[INFO] |  +- org.datanucleus:datanucleus-core:jar:4.1.17:compile
[INFO] |  +- org.apache.hadoop:hadoop-client-runtime:jar:3.3.4:compile
[INFO] |  +- org.apache.thrift:libthrift:jar:0.12.0:compile
[INFO] |  +- org.apache.thrift:libfb303:jar:0.9.3:compile
[INFO] |  \- org.apache.derby:derby:jar:10.14.2.0:compile
[INFO] +- org.apache.spark:spark-core_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] |  +- com.twitter:chill_2.13:jar:0.10.0:compile
[INFO] |  |  \- com.esotericsoftware:kryo-shaded:jar:4.0.2:compile
[INFO] |  |     \- com.esotericsoftware:minlog:jar:1.3.0:compile
[INFO] |  +- com.twitter:chill-java:jar:0.10.0:compile
[INFO] |  +- org.apache.xbean:xbean-asm9-shaded:jar:4.21:compile
[INFO] |  +- org.apache.hadoop:hadoop-client-api:jar:3.3.4:compile
[INFO] |  +- org.apache.spark:spark-launcher_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-kvstore_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  |  +- org.fusesource.leveldbjni:leveldbjni-all:jar:1.8:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.13.4:compile
[INFO] |  +- org.apache.spark:spark-network-common_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  |  \- com.google.crypto.tink:tink:jar:1.7.0:compile
[INFO] |  +- org.apache.spark:spark-network-shuffle_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.spark:spark-unsafe_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- javax.activation:activation:jar:1.1.1:compile
[INFO] |  +- org.apache.curator:curator-recipes:jar:2.13.0:compile
[INFO] |  |  \- org.apache.curator:curator-framework:jar:2.13.0:compile
[INFO] |  |     \- org.apache.curator:curator-client:jar:2.13.0:compile
[INFO] |  +- org.apache.zookeeper:zookeeper:jar:3.6.2:compile
[INFO] |  |  +- org.apache.zookeeper:zookeeper-jute:jar:3.6.2:compile
[INFO] |  |  \- org.apache.yetus:audience-annotations:jar:0.5.0:compile
[INFO] |  +- jakarta.servlet:jakarta.servlet-api:jar:4.0.3:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.12.0:compile
[INFO] |  +- org.apache.commons:commons-math3:jar:3.6.1:compile
[INFO] |  +- org.apache.commons:commons-text:jar:1.9:compile
[INFO] |  +- commons-io:commons-io:jar:2.11.0:compile
[INFO] |  +- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] |  +- org.apache.commons:commons-collections4:jar:4.4:compile
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.36:compile
[INFO] |  +- org.slf4j:jul-to-slf4j:jar:1.7.36:compile
[INFO] |  +- org.slf4j:jcl-over-slf4j:jar:1.7.36:compile
[INFO] |  +- org.apache.logging.log4j:log4j-slf4j-impl:jar:2.19.0:compile
[INFO] |  +- org.apache.logging.log4j:log4j-api:jar:2.19.0:compile
[INFO] |  +- org.apache.logging.log4j:log4j-core:jar:2.19.0:compile
[INFO] |  +- org.apache.logging.log4j:log4j-1.2-api:jar:2.19.0:compile
[INFO] |  +- com.ning:compress-lzf:jar:1.1:compile
[INFO] |  +- org.xerial.snappy:snappy-java:jar:1.1.8.4:compile
[INFO] |  +- org.lz4:lz4-java:jar:1.8.0:compile
[INFO] |  +- com.github.luben:zstd-jni:jar:1.5.2-3:compile
[INFO] |  +- org.roaringbitmap:RoaringBitmap:jar:0.9.32:compile
[INFO] |  |  \- org.roaringbitmap:shims:jar:0.9.32:runtime
[INFO] |  +- org.scala-lang.modules:scala-xml_2.13:jar:2.1.0:compile
[INFO] |  +- org.scala-lang:scala-library:jar:2.13.8:compile
[INFO] |  +- org.scala-lang:scala-reflect:jar:2.13.8:compile
[INFO] |  +- org.json4s:json4s-jackson_2.13:jar:3.7.0-M11:compile
[INFO] |  |  \- org.json4s:json4s-core_2.13:jar:3.7.0-M11:compile
[INFO] |  |     +- org.json4s:json4s-ast_2.13:jar:3.7.0-M11:compile
[INFO] |  |     \- org.json4s:json4s-scalap_2.13:jar:3.7.0-M11:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-client:jar:2.36:compile
[INFO] |  |  +- jakarta.ws.rs:jakarta.ws.rs-api:jar:2.1.6:compile
[INFO] |  |  \- org.glassfish.hk2.external:jakarta.inject:jar:2.6.1:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-common:jar:2.36:compile
[INFO] |  |  +- jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile
[INFO] |  |  \- org.glassfish.hk2:osgi-resource-locator:jar:1.0.3:compile
[INFO] |  +- org.glassfish.jersey.core:jersey-server:jar:2.36:compile
[INFO] |  |  \- jakarta.validation:jakarta.validation-api:jar:2.0.2:compile
[INFO] |  +- org.glassfish.jersey.containers:jersey-container-servlet:jar:2.36:compile
[INFO] |  +- org.glassfish.jersey.containers:jersey-container-servlet-core:jar:2.36:compile
[INFO] |  +- org.glassfish.jersey.inject:jersey-hk2:jar:2.36:compile
[INFO] |  |  +- org.glassfish.hk2:hk2-locator:jar:2.6.1:compile
[INFO] |  |  |  +- org.glassfish.hk2.external:aopalliance-repackaged:jar:2.6.1:compile
[INFO] |  |  |  +- org.glassfish.hk2:hk2-api:jar:2.6.1:compile
[INFO] |  |  |  \- org.glassfish.hk2:hk2-utils:jar:2.6.1:compile
[INFO] |  |  \- org.javassist:javassist:jar:3.25.0-GA:compile
[INFO] |  +- io.netty:netty-all:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-buffer:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-codec:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-common:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-handler:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-transport-native-unix-common:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-resolver:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-transport:jar:4.1.80.Final:compile
[INFO] |  |  +- io.netty:netty-transport-classes-epoll:jar:4.1.80.Final:compile
[INFO] |  |  \- io.netty:netty-transport-classes-kqueue:jar:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-transport-native-epoll:jar:linux-aarch_64:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-transport-native-kqueue:jar:osx-aarch_64:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.80.Final:compile
[INFO] |  +- io.netty:netty-tcnative-classes:jar:2.0.54.Final:compile
[INFO] |  +- com.clearspring.analytics:stream:jar:2.9.6:compile
[INFO] |  +- io.dropwizard.metrics:metrics-core:jar:4.2.10:compile
[INFO] |  +- io.dropwizard.metrics:metrics-jvm:jar:4.2.10:compile
[INFO] |  +- io.dropwizard.metrics:metrics-json:jar:4.2.10:compile
[INFO] |  +- io.dropwizard.metrics:metrics-graphite:jar:4.2.10:compile
[INFO] |  +- io.dropwizard.metrics:metrics-jmx:jar:4.2.10:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.13.4:compile
[INFO] |  +- com.fasterxml.jackson.module:jackson-module-scala_2.13:jar:2.13.4:compile
[INFO] |  |  \- com.thoughtworks.paranamer:paranamer:jar:2.8:compile
[INFO] |  +- org.apache.ivy:ivy:jar:2.5.0:compile
[INFO] |  +- oro:oro:jar:2.0.8:compile
[INFO] |  +- net.razorvine:pickle:jar:1.2:compile
[INFO] |  +- net.sf.py4j:py4j:jar:0.10.9.7:compile
[INFO] |  \- org.apache.commons:commons-crypto:jar:1.1.0:compile
[INFO] +- org.apache.spark:spark-catalyst_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] |  +- org.scala-lang.modules:scala-parser-combinators_2.13:jar:1.1.2:compile
[INFO] |  +- org.apache.spark:spark-sketch_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.codehaus.janino:janino:jar:3.1.7:compile
[INFO] |  +- org.codehaus.janino:commons-compiler:jar:3.1.7:compile
[INFO] |  +- org.antlr:antlr4-runtime:jar:4.9.3:compile
[INFO] |  +- com.univocity:univocity-parsers:jar:2.9.1:compile
[INFO] |  +- org.apache.arrow:arrow-vector:jar:9.0.0:compile
[INFO] |  |  +- org.apache.arrow:arrow-format:jar:9.0.0:compile
[INFO] |  |  +- org.apache.arrow:arrow-memory-core:jar:9.0.0:compile
[INFO] |  |  +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.13.4:compile
[INFO] |  |  \- com.google.flatbuffers:flatbuffers-java:jar:1.12.0:compile
[INFO] |  \- org.apache.arrow:arrow-memory-netty:jar:9.0.0:compile
[INFO] +- org.apache.spark:spark-hive_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] +- org.scala-lang.modules:scala-parallel-collections_2.13:jar:1.0.3:compile
[INFO] +- com.google.guava:guava:jar:14.0.1:provided
[INFO] +- org.apache.hive:hive-cli:jar:2.3.9:compile
[INFO] |  +- commons-lang:commons-lang:jar:2.6:compile
[INFO] |  \- jline:jline:jar:2.14.6:compile
[INFO] +- org.apache.hive:hive-jdbc:jar:2.3.9:compile
[INFO] |  \- org.apache.httpcomponents:httpcore:jar:4.4.14:compile
[INFO] +- org.apache.hive:hive-beeline:jar:2.3.9:compile
[INFO] |  \- net.sf.supercsv:super-csv:jar:2.2.0:compile
[INFO] +- org.apache.hive:hive-service-rpc:jar:3.1.3:compile
[INFO] +- org.eclipse.jetty:jetty-server:jar:9.4.48.v20220622:provided
[INFO] |  +- javax.servlet:javax.servlet-api:jar:3.1.0:provided
[INFO] |  +- org.eclipse.jetty:jetty-http:jar:9.4.48.v20220622:provided
[INFO] |  |  \- org.eclipse.jetty:jetty-util:jar:9.4.48.v20220622:provided
[INFO] |  \- org.eclipse.jetty:jetty-io:jar:9.4.48.v20220622:provided
[INFO] +- org.eclipse.jetty:jetty-servlet:jar:9.4.48.v20220622:provided
[INFO] |  +- org.eclipse.jetty:jetty-security:jar:9.4.48.v20220622:provided
[INFO] |  \- org.eclipse.jetty:jetty-util-ajax:jar:9.4.48.v20220622:provided
[INFO] +- org.seleniumhq.selenium:selenium-java:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-api:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-chrome-driver:jar:4.2.2:test
[INFO] |  |  +- org.seleniumhq.selenium:selenium-chromium-driver:jar:4.2.2:test
[INFO] |  |  \- org.seleniumhq.selenium:selenium-json:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-devtools-v100:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-devtools-v101:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-devtools-v102:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-devtools-v85:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-edge-driver:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-firefox-driver:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-ie-driver:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-opera-driver:jar:4.2.2:test
[INFO] |  +- org.seleniumhq.selenium:selenium-remote-driver:jar:4.2.2:test
[INFO] |  |  +- com.beust:jcommander:jar:1.82:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-api:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-context:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-exporter-logging:jar:1.14.0:test
[INFO] |  |  |  +- io.opentelemetry:opentelemetry-sdk-metrics:jar:1.14.0:test
[INFO] |  |  |  \- io.opentelemetry:opentelemetry-sdk-logs:jar:1.14.0-alpha:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk-common:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk-extension-autoconfigure:jar:1.14.0-alpha:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk-trace:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-sdk:jar:1.14.0:test
[INFO] |  |  +- io.opentelemetry:opentelemetry-semconv:jar:1.14.0-alpha:test
[INFO] |  |  +- io.ous:jtoml:jar:2.0.0:test
[INFO] |  |  +- org.apache.commons:commons-exec:jar:1.3:test
[INFO] |  |  +- org.asynchttpclient:async-http-client:jar:2.12.3:test
[INFO] |  |  |  +- org.asynchttpclient:async-http-client-netty-utils:jar:2.12.3:test
[INFO] |  |  |  +- org.reactivestreams:reactive-streams:jar:1.0.3:test
[INFO] |  |  |  +- com.typesafe.netty:netty-reactive-streams:jar:2.0.4:test
[INFO] |  |  |  \- com.sun.activation:jakarta.activation:jar:1.2.2:test
[INFO] |  |  \- org.seleniumhq.selenium:selenium-http:jar:4.2.2:test
[INFO] |  |     \- dev.failsafe:failsafe:jar:3.2.4:test
[INFO] |  +- org.seleniumhq.selenium:selenium-safari-driver:jar:4.2.2:test
[INFO] |  \- org.seleniumhq.selenium:selenium-support:jar:4.2.2:test
[INFO] +- org.seleniumhq.selenium:htmlunit-driver:jar:3.62.0:test
[INFO] |  \- net.sourceforge.htmlunit:htmlunit:jar:2.62.0:test
[INFO] |     +- xalan:xalan:jar:2.7.2:test
[INFO] |     |  \- xalan:serializer:jar:2.7.2:test
[INFO] |     +- org.apache.httpcomponents:httpmime:jar:4.5.13:test
[INFO] |     +- net.sourceforge.htmlunit:htmlunit-core-js:jar:2.62.0:test
[INFO] |     +- net.sourceforge.htmlunit:neko-htmlunit:jar:2.62.0:test
[INFO] |     |  \- xerces:xercesImpl:jar:2.12.2:test
[INFO] |     |     \- xml-apis:xml-apis:jar:1.4.01:test
[INFO] |     +- net.sourceforge.htmlunit:htmlunit-cssparser:jar:1.12.0:test
[INFO] |     +- commons-net:commons-net:jar:3.8.0:test
[INFO] |     +- org.brotli:dec:jar:0.1.2:test
[INFO] |     +- com.shapesecurity:salvation2:jar:3.0.0:test
[INFO] |     \- org.eclipse.jetty.websocket:websocket-client:jar:9.4.46.v20220331:test
[INFO] |        +- org.eclipse.jetty:jetty-client:jar:9.4.48.v20220622:provided
[INFO] |        \- org.eclipse.jetty.websocket:websocket-common:jar:9.4.46.v20220331:test
[INFO] |           \- org.eclipse.jetty.websocket:websocket-api:jar:9.4.46.v20220331:test
[INFO] +- org.apache.spark:spark-sql_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] |  +- org.rocksdb:rocksdbjni:jar:7.5.3:compile
[INFO] |  +- org.apache.spark:spark-catalyst_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] |  +- org.apache.orc:orc-core:jar:shaded-protobuf:1.8.0:compile
[INFO] |  |  +- org.apache.orc:orc-shims:jar:1.8.0:compile
[INFO] |  |  +- io.airlift:aircompressor:jar:0.21:compile
[INFO] |  |  +- org.jetbrains:annotations:jar:17.0.0:compile
[INFO] |  |  \- org.threeten:threeten-extra:jar:1.7.1:compile
[INFO] |  +- org.apache.orc:orc-mapreduce:jar:shaded-protobuf:1.8.0:compile
[INFO] |  +- org.apache.hive:hive-storage-api:jar:2.7.3:compile
[INFO] |  +- org.apache.parquet:parquet-column:jar:1.12.3:compile
[INFO] |  |  +- org.apache.parquet:parquet-common:jar:1.12.3:compile
[INFO] |  |  \- org.apache.parquet:parquet-encoding:jar:1.12.3:compile
[INFO] |  \- org.apache.parquet:parquet-hadoop:jar:1.12.3:compile
[INFO] |     +- org.apache.parquet:parquet-format-structures:jar:1.12.3:compile
[INFO] |     \- org.apache.parquet:parquet-jackson:jar:1.12.3:runtime
[INFO] +- org.apache.spark:spark-tags_2.13:jar:3.4.0-SNAPSHOT:compile
[INFO] +- org.apache.spark:spark-tags_2.13:test-jar:tests:3.4.0-SNAPSHOT:test
[INFO] +- org.mockito:mockito-core:jar:4.6.1:test
[INFO] |  +- net.bytebuddy:byte-buddy:jar:1.12.10:test
[INFO] |  +- net.bytebuddy:byte-buddy-agent:jar:1.12.10:test
[INFO] |  \- org.objenesis:objenesis:jar:3.2:compile
[INFO] +- net.sf.jpam:jpam:jar:1.1:compile
[INFO] |  \- commons-logging:commons-logging:jar:1.1.3:compile
[INFO] +- commons-cli:commons-cli:jar:1.5.0:compile
[INFO] +- org.spark-project.spark:unused:jar:1.0.0:compile
[INFO] +- org.scalatest:scalatest_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-core_2.13:jar:3.2.13:test
[INFO] |  |  +- org.scalatest:scalatest-compatible:jar:3.2.13:test
[INFO] |  |  \- org.scalactic:scalactic_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-featurespec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-flatspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-freespec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-funsuite_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-funspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-propspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-refspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-wordspec_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-diagrams_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-matchers-core_2.13:jar:3.2.13:test
[INFO] |  +- org.scalatest:scalatest-shouldmatchers_2.13:jar:3.2.13:test
[INFO] |  \- org.scalatest:scalatest-mustmatchers_2.13:jar:3.2.13:test
[INFO] +- org.scalatestplus:scalacheck-1-16_2.13:jar:3.2.13.0:test
[INFO] |  \- org.scalacheck:scalacheck_2.13:jar:1.16.0:test
[INFO] +- org.scalatestplus:mockito-4-6_2.13:jar:3.2.13.0:test
[INFO] +- org.scalatestplus:selenium-4-2_2.13:jar:3.2.13.0:test
[INFO] +- junit:junit:jar:4.13.2:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] \- com.github.sbt:junit-interface:jar:0.13.3:test
[INFO]    \- org.scala-sbt:test-interface:jar:1.0:test
```

Reply:
Is this a bug of  `maven-dependency-plugin 3.3.0`? Does 3.2.x have the same problem?

Reply:
run `mvn dependency:copy-dependencies`, I found `httpcore-4.4.14.jar` also copied to `target/dependency` directory.

Reply:
Yes, mvn run on local is same!
But when you use **sbt** and run `Scala 2.13 build with SBT` on github, build classpath will show different.
The build class path above is extracted from github's workflow log.
<img width="825" alt="image" src="https://user-images.githubusercontent.com/15246973/193269919-d41826fd-fc5a-45e4-9be1-500dfb74b605.png">

Reply:
> Is this a bug of `maven-dependency-plugin 3.3.0`? Does 3.2.x have the same problem?

version 3.2.0 is ok when i run it on github's workflow.
But I don't think it is a bug about 3.3.0, because explicit import the dependency is also reasonable.
The code in hive-thriftserver directly uses the above class library(httpcore)

Reply:
> run `mvn dependency:copy-dependencies`, I found `httpcore-4.4.14.jar` also copied to `target/dependency` directory.
> 
> @panbingkun Cloud we find a version without issue? In addition, why does this not affect the compilation of Scala 2.12? I'm worried about other hidden dangers in this version

without httpcore-4.4.13.jar

Reply:
I think the upgrade is acceptable, but I still recommend using 3.2.x first, and then waiting for the next version

Reply:
ok, let’s wait more eys on it.

Reply:
If it's an sbt vs Maven difference, I'm not as concerned. It sounds like a real issue, like, the hive-thriftserver code really does directly depend on this. In that case this is a good fix, if anything. I'm OK with the change going to 3.3.0.